### PR TITLE
Upgrade docker version images

### PR DESCRIPTION
**Pull request by pjgg**
Keycloak -> 15.0.2

Strimzi/Apicurio is updated in Quarkus-test-framework: https://github.com/quarkus-qe/quarkus-test-framework/pull/272

databases:
```
Baremetal:
postgresql -> quay.io/bitnami/postgresql:13.4.0
mariaDB -> quay.io/quarkusqeteam/mariadb:10.6.4 
mysql -> quay.io/bitnami/mysql:5.7.32
DB2 and mssql leave it as it is 

openshift
postgresql -> registry.access.redhat.com/rhscl/postgresql-10-rhel7
                        registry.redhat.io/rhscl/postgresql-12-rhel7
mariaDB -> registry.redhat.io/rhscl/mariadb-103-rhel7
                    registry.access.redhat.com/rhscl/mariadb-102-rhel7
mysql -> registry.access.redhat.com/rhscl/mysql-80-rhel7
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -199,12 +199,12 @@
...
                                 <!-- Services used in test suite -->
-                                <keycloak.image>quay.io/keycloak/keycloak:14.0.0</keycloak.image>
+                                <keycloak.image>quay.io/keycloak/keycloak:15.0.2</keycloak.image>
                                 <rhsso.73.image>registry.access.redhat.com/redhat-sso-7/sso73-openshift</rhsso.73.image>
-                                <postgresql.10.image>registry.access.redhat.com/rhscl/postgresql-10-rhel7</postgresql.10.image>
+                                <postgresql.13.image>quay.io/bitnami/postgresql:13.4.0</postgresql.13.image>
                                 <mysql.57.image>quay.io/bitnami/mysql:5.7.32</mysql.57.image>
                                 <mysql.80.image>registry.access.redhat.com/rhscl/mysql-80-rhel7</mysql.80.image>
-                                <mariadb.102.image>registry.access.redhat.com/rhscl/mariadb-102-rhel7</mariadb.102.image>
+                                <mariadb.1064.image>quay.io/quarkusqeteam/mariadb:10.6.4</mariadb.1064.image>
```
Comment (Sgitario) [inline pom.xml:207]: I would prefer simply `mariadb.10.6.4`.

**Review by mjurc — APPROVED** [review]
LGTM

**Review by Sgitario — CHANGES_REQUESTED** [review]
Please, update the `sql-db/sql-app` section in the README too.
Pending to clarification that we're losing the redhat images coverage.

**Review by Sgitario — APPROVED** [review]

**Review by mjurc — CHANGES_REQUESTED** [review]
Actually, thinking on it now - did we not agree to keep coverage for both latest upstream and RH scl images?

**Comment by pjgg** [comment]
> Actually, thinking on it now - did we not agree to keep coverage for both latest upstream and RH scl images?

The only RH image that I have changed is MariaDB, but Is in bare metal. I thought that we will use RH images only in Openshift-ts but we are not going to use the latest one yet because some of them require authentication, and we have to double-check if the TestFW supports it.

**Comment by Sgitario** [comment]
The mariadb image in quay.io was not public. I've changed it and retriggered the CI.

**Comment by Sgitario** [comment]
run tests

**Comment by mjurc** [comment]
run tests

**Review by Sgitario — CHANGES_REQUESTED** [review]
The tests using MariaDB:10.6.4 are not working in OpenShift:

```
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] mariadbd: Can't create/write to file '/var/lib/mysql/aria_log_control' (Errcode: 13 "Permission denied")&amp#27;[0m
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] mariadbd: Got error 'Can't create file' when trying to use aria control file '/var/lib/mysql/aria_log_control'&amp#27;[0m
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] Plugin 'Aria' init function returned error.&amp#27;[0m
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] Plugin 'Aria' registration as a STORAGE ENGINE failed.&amp#27;[0m
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] InnoDB: Operating system error number 13 in a file operation.&amp#27;[0m
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] InnoDB: The error means mariadbd does not have the access rights to the directory.&amp#27;[0m
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] InnoDB: Operating system error number 13 in a file operation.&amp#27;[0m
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] InnoDB: The error means mariadbd does not have the access rights to the directory.&amp#27;[0m
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] InnoDB: Cannot open datafile './ibdata1'&amp#27;[0m
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] InnoDB: Could not open or create the system tablespace. If you tried to add new data files to the system tablespace, and it failed here, you should now edit innodb_data_file_path in my.cnf back to what it was, and remove the new ibdata files InnoDB created in this failed attempt. InnoDB only wrote those files full of zeros, but did not yet use them in any way. But be careful: do not remove old data files which contain your precious data!&amp#27;[0m
01:25:30.519 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] InnoDB: Database creation was aborted with error Cannot open a file. You may need to delete the ibdata1 file before trying to start up again.&amp#27;[0m
01:25:30.520 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] Plugin 'InnoDB' init function returned error.&amp#27;[0m
01:25:30.520 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.&amp#27;[0m
01:25:30.520 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] Failed to initialize plugins.&amp#27;[0m
01:25:30.520 INFO  &amp#27;[0;34m[database] 2021-09-23  1:25:28 0 [ERROR] Aborting&amp#27;[0m
```

This is affecting OpenShift tests using the new Maria DB: quay.io/quarkusqeteam/mariadb:10.6.4. We need to investigate why this image is incompatible with OpenShift before merging this PR.

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Review by Sgitario — CHANGES_REQUESTED** [review]
There are still failures on OpenShift: https://main-jenkins-csb-quarkusqe.apps.ocp4.prod.psi.redhat.com/job/quarkus-main-rhel8-jdk11-openshift-new-ts-jvm/114/testReport/

Some failures are related to these changes: https://main-jenkins-csb-quarkusqe.apps.ocp4.prod.psi.redhat.com/job/quarkus-main-rhel8-jdk11-openshift-new-ts-jvm/114/testReport/io.quarkus.ts.spring.web.boostrap/OpenShiftBookResourceIT/io_quarkus_ts_spring_web_boostrap_OpenShiftBookResourceIT/

**Comment by pjgg** [comment]
run tests

**Review by Sgitario — APPROVED** [review]

**Review by mjurc — APPROVED** [review]

**Pull request merged**